Bearded.Graphics.ImageSharp/ImageTextureData.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Bearded.Graphics.Textures;
+using OpenTK.Graphics.OpenGL;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace Bearded.Graphics.ImageSharp;
+
+public sealed class ImageTextureData : ITextureData
+{
+    private static readonly Configuration configuration;
+
+    static ImageTextureData()
+    {
+        configuration = Configuration.Default.Clone();
+        configuration.PreferContiguousImageBuffers = true;
+    }
+
+    private readonly Image<Bgra32> image;
+
+    public int Width { get; }
+
+    public int Height { get; }
+
+    public static ITextureData From(string path) => new ImageTextureData(Image.Load<Bgra32>(configuration, path));
```

Comment:
Out of curiosity: is the choice to return an `ITextureData` here on purpose? While I like hiding the implementation details, it is also somewhat unintuitive that any public fields added to this type will never be accessible without explicit casting (but they _are_ accessible 😛)

Reply:
'Hiding' the implementation was on purpose, that being said, there are no public members that don't implement the interface right now, so I don't see a problem here.

Except, I suppose we could expose the internal image to allow modification and forward the Width and Height properties. But I'm happy to make that (slightly) breaking signature change when/if we need it and keep it a little cleaner for now.